wagtail/admin/views/reports/page_types_usage.py

```diff
@@ -0,0 +1,191 @@
+from wagtail.coreutils import get_content_languages
+from wagtail.models import ContentType, Page, PageLogEntry, Site, get_page_models
+from wagtail.users.utils import get_deleted_user_display_name
+
+
+def _get_locale_choices():
+    choices = [
+        (language_code, display_name)
+        for language_code, display_name in get_content_languages().items()
+    ]
+    return choices
+
+
+def _get_site_choices():
+    choices = [(site.pk, str(site)) for site in Site.objects.all()]
+    return choices
+
+
+def _annotate_last_edit_info(queryset):
+    latest_edit_log = PageLogEntry.objects.filter(content_type=OuterRef("pk")).order_by(
+        "-timestamp", "-pk"
+    )
+
+    queryset = queryset.annotate(
+        count=Count("pages"),
+        last_edited_page=Subquery(latest_edit_log.values("page")[:1]),
+        last_edited_by=Subquery(latest_edit_log.values("user")[:1]),
+    )
+
+    return queryset
+
+
+def _annotate_last_edit_info_by_locale(queryset, language_code):
+    latest_edit_log = PageLogEntry.objects.filter(
+        content_type=OuterRef("pk"), page__locale__language_code=language_code
+    ).order_by("-timestamp", "-pk")
+
+    queryset = queryset.annotate(
+        count=Count("pages", filter=Q(pages__locale__language_code=language_code)),
+        last_edited_page=Subquery(latest_edit_log.values("page")[:1]),
+        last_edited_by=Subquery(latest_edit_log.values("user")[:1]),
+    )
+
+    return queryset
+
+
+class LocaleFilter(django_filters.ChoiceFilter):
+    def filter(self, qs, value):
+        # The filtering is handled in the filter_queryset method
+        return qs
+
+
+class SiteFilter(django_filters.ChoiceFilter):
+    def filter(self, qs, value):
+        # The filtering is handled in the filter_queryset method
+        return qs
+
+
+class PageTypesUsageReportFilterSet(WagtailFilterSet):
+    page_locale = LocaleFilter(
+        label=_("Locale"),
+        choices=_get_locale_choices,
+        empty_label=None,
+        null_label=_("All"),
+        null_value="all",
+    )
+    site = SiteFilter(
+        label=_("Site"),
+        choices=_get_site_choices,
+        empty_label=None,
+        null_label=_("All"),
+        null_value="all",
+    )
+
+    class Meta:
+        model = ContentType
+        fields = ["page_locale", "site"]
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self.sites = {
+            site.pk: site for site in Site.objects.all().prefetch_related("root_page")
+        }
+        self.sites_filter_enabled = True
+        if len(self.sites) == 1:
+            # If there is only one site, we don't need to show the site filter
+            self.sites_filter_enabled = False
+            del self.filters["site"]
+
+    def filter_queryset(self, queryset):
+        queryset = super().filter_queryset(queryset)
+
+        if self.sites_filter_enabled:
+            site = self.form.cleaned_data["site"]
```

Comment:
The `if self.sites_filter_enabled` is required because it disables the `site` filter field when there is only one site on the build. In those cases, it doesn't make sense to show a filter option, IMO